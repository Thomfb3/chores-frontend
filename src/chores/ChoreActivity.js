import React, { useEffect, useState, useContext }from "react";
import findUserInTeam from "../helpers/findUserInTeam";
import formatDate from "../helpers/formatDate";
import UserContext from "../auth/UserContext";


function ChoreActivity({user, event, date}) { 
    const { currentUser, currentTeamUsers } = useContext(UserContext);

    const choreActivityDate = formatDate(date);
    const choreActivityUser = findUserInTeam(user, currentTeamUsers);

    return (
        <div>
            <p>{choreActivityUser.username}</p>
            <p>{event}</p>
            <p>{choreActivityDate}</p>
        </div>
    );
}


export default ChoreActivity