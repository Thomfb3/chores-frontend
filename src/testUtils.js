import React from "react";
import UserContext from "./auth/UserContext";
import ChoreCommentContext from "./comments/ChoreCommentContext";

const demoUser = {
  _id: "userid1",
  username: "testuser1",
  firstName: "testfirst1",
  lastName: "testlast1",
  profileImage: "defaultProfile.jpg",
  allTimePoints: 1000,
  currentPoints: 200,
  role: "admin"
};

const testTeam = {
  _id: "testTeam",
  name: "testTeam"
};

const testTeamUsers = [
  {
    _id: "userid1",
    username: "testuser1",
    firstName: "testfirst1",
    lastName: "testlast1",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "admin"
  },
  {
    _id: "userid2",
    username: "testuser2",
    firstName: "testfirst2",
    lastName: "testlast2",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid3",
    username: "testuser3",
    firstName: "testfirst3",
    lastName: "testlast3",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid4",
    username: "testuser4",
    firstName: "testfirst4",
    lastName: "testlast4",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  }
];

let choreComments = [];

const UserProvider =
  ({ children,
    currentUser = demoUser,
    currentTeam = testTeam,
    currentTeamUsers = testTeamUsers,
    isAdmin = false
  }) => (
      <UserContext.Provider value={{ currentUser, currentTeam, currentTeamUsers }}>
        {children}
      </UserContext.Provider>
    );


const CommentProvider =
  ({ children,
    currentChoreComments = choreComments
  }) => (
      <ChoreCommentContext.Provider value={{ currentChoreComments }}>
        {children}
      </ChoreCommentContext.Provider>
    );

export { UserProvider, CommentProvider };

