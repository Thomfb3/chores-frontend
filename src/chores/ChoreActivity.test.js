import React from "react";
import { render } from "@testing-library/react";
import ChoreActivity from "./ChoreActivity";
import { MemoryRouter, Route } from "react-router-dom";
import { UserProvider, CommentProvider } from "../testUtils";

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <UserProvider>
                <CommentProvider>
                    <ChoreActivity 
                        key="username"
                        user="username"
                        event="chore created"
                        status="Created"
                        date="2021-11-11T19:35:37.000+00:00"
                    />
                </CommentProvider>
            </UserProvider>
        </MemoryRouter>,
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter initialEntries={["/chore/123"]}>
            <UserProvider>
                <Route path="/chore/:handle">
                    <CommentProvider>
                    <ChoreActivity 
                        key="username"
                        user="username"
                        event="chore created"
                        status="Created"
                        date="2021-11-11T19:35:37.000+00:00"
                    />
                    </CommentProvider>
                </Route>
            </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});
