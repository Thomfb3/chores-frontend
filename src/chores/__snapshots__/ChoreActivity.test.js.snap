// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ChoreActivity"
  >
    <div
      class="ChoreActivity__status-marker ChoreActivity__status-marker--Created"
    >
      Created
    </div>
    <p
      class="ChoreActivity__event"
    >
      Chore created - 
    </p>
    <p
      class="ChoreActivity__event"
    >
      by undefined - 
    </p>
    <p
      class="ChoreActivity__date"
    >
      on 11/11/2021 @ 2:35 PM
    </p>
  </div>
</DocumentFragment>
`;
